src/validators/DefaultValidator.ts

```diff
@@ -0,0 +1,26 @@
+import type { IConstraint } from '../constraints/base/IConstraint';
+import type { ValidationError } from '../lib/errors/ValidationError';
+import { Result } from '../lib/Result';
+import { BaseValidator } from './imports';
+import { getValue } from './util/getValue';
+
+export class DefaultValidator<T> extends BaseValidator<T> {
+	private readonly validator: BaseValidator<T>;
+	private readonly defaultValue: T | (() => T);
+
+	public constructor(validator: BaseValidator<T>, value: T | (() => T), constraints: readonly IConstraint<T>[] = []) {
+		super(constraints);
+		this.validator = validator;
+		this.defaultValue = value;
+	}
+
+	protected handle(value: undefined): Result<T, ValidationError | AggregateError> {
```

Comment:
```suggestion
	protected handle(value: T | undefined): Result<T, ValidationError | AggregateError> {
```

Reply:
This was actually a mistype, I meant to put unknown instead of undefined, should I change it to unknown to match every other validator or use this suggestion?

Reply:
unknown, I updated the suggestion.